packages/git/src/browser/diff/git-diff-widget.ts

```diff
@@ -0,0 +1,148 @@
+            this.options = options;
+            await this.updateView();
+        }
+    }
+
+    protected async updateView() {
+        const commitishBar = await this.renderDiffListHeader();
+        if (this.options && this.options.fromRevision && this.options.toRevision) {
+            const fromRevision = this.options.fromRevision.toString();
+            const toRevision = this.options.toRevision;
+            const fileChangeList = await this.renderFileChangeList(this.fileChanges, toRevision, fromRevision);
+            this.dom = h.div({ className: "git-diff-container" }, VirtualRenderer.flatten([commitishBar, fileChangeList]));
+            this.update();
+        }
+    }
+
+    protected render(): h.Child {
+        return this.dom;
+    }
+
+    protected async renderDiffListHeader(): Promise<h.Child> {
+        if (this.options) {
+            let fileDiv: h.Child = '';
+            const header = this.options.title ? h.div({ className: 'git-diff-header' }, this.options.title) : '';
+            if (this.options.fileUri) {
+                const repository = this.repositoryProvider.selectedRepository;
+                const [icon, label, path] = await Promise.all([
+                    this.labelProvider.getIcon(this.options.fileUri),
+                    this.labelProvider.getName(this.options.fileUri),
+                    repository ? GitUtils.getRepositoryRelativePath(repository, this.options.fileUri) : this.labelProvider.getLongName(this.options.fileUri)
+                ]);
+                const iconSpan = h.span({ className: icon + ' file-icon' });
+                const nameSpan = h.span({ className: 'name' }, label + ' ');
+                const pathSpan = h.span({ className: 'path' }, path);
+                const compareDiv = h.div({ className: 'theia-header' }, 'Compare...');
+                fileDiv = h.div({ className: "gitItem diff-file" }, h.div({ className: "noWrapInfo" }, iconSpan, nameSpan, pathSpan));
+                const inSpan = h.span({ className: 'row-title' }, 'in:');
+                const withSpan = h.span({ className: 'row-title' }, 'with:');
+                const toDiv = this.options.toRevision ? h.div({ className: "revision noWrapInfo" }, inSpan, this.options.toRevision) : '';
+                const fromDiv = this.options.fromRevision ? h.div({ className: "revision noWrapInfo" }, withSpan, this.options.fromRevision.toString()) : '';
+                return h.div({ className: "commitishBar" }, header, compareDiv, fileDiv, toDiv, fromDiv);
+            }
+            return h.div({ className: "commitishBar" }, header);
+        }
+        return '';
+    }
+
+    protected async renderFileChangeList(fileChanges: GitFileChange[], commitSha: string, toCommitSha?: string): Promise<h.Child> {
+        const files: h.Child[] = [];
+
+        for (const fileChange of fileChanges) {
+            const fileChangeElement: h.Child = await this.renderGitItem(fileChange, commitSha, toCommitSha);
```

Comment:
[Just like above](https://github.com/theia-ide/theia/pull/1092/files#r163005170); the `toCommitSha` is treated as `fromCommitSha` when you call `renderGitItem`. We shall check this too.